docker/Dockerfile

```diff
@@ -2,36 +2,48 @@
 # Install OpenSearch Benchmark to build a Docker image #
 ########################################################
 
-ARG VERSION
-ARG BUILD_ENV=production
+#
+# Stage 1: build packages and compile where needed
+#
+FROM python:3.11.2-slim AS build-stage
+WORKDIR /opensearch-benchmark
+
+RUN apt-get -y update && \
+    apt-get install -y curl git gcc pbzip2 pigz make jq && \
+    apt-get -y upgrade
+
+COPY . opensearch-benchmark
 
-FROM python:3.11.2-slim as build_env_testing
-ONBUILD COPY opensearch-benchmark-git/ ./
+RUN cd opensearch-benchmark; make wheel
 
-FROM python:3.11.2-slim as build_env_production
-ONBUILD RUN echo Production Environment
 
-FROM build_env_${BUILD_ENV}
+#
+# Stage 2: create image
+#
+FROM python:3.11.2-slim AS image-stage
 WORKDIR /opensearch-benchmark
 ENV BENCHMARK_RUNNING_IN_DOCKER=True
 
-RUN apt-get -y update && \
-    apt-get install -y curl git gcc pbzip2 pigz && \
-    apt-get -y upgrade && \
-    rm -rf /var/lib/apt/lists/*
-
 RUN groupadd --gid 1000 opensearch-benchmark && \
     useradd -d /opensearch-benchmark -m -k /dev/null -g 1000 -N -u 1000 -l -s /bin/bash benchmark
 
-ENV PIP_ONLY_BINARY=h5py
-RUN if [ "$BUILD_ENV" = "testing" ] ; then echo Testing; ls -l; python3 -m pip install -e . ; \
-    else echo Production; if [ -z "$VERSION" ] ; then python3 -m pip install opensearch-benchmark ; else python3 -m pip install opensearch-benchmark==$VERSION ; fi; fi
-
 RUN mkdir -p /opensearch-benchmark/.benchmark && \
     chown -R 1000:0 /opensearch-benchmark/.benchmark
 
+COPY --from=build-stage /opensearch-benchmark/opensearch-benchmark/yappi-*.whl /opensearch-benchmark/opensearch-benchmark/opensearch_benchmark-*.whl ./
+
+# There is no binary package currently available for yappi on ARM.
```

Comment:
Will this cause issue on arm64 based on your comment?

Since we release multi-arch image to the public.

Thanks.

Reply:
No, this is handled by copying the locally-built `yappi` wheel from the first stage, so the C compiler does not need to be installed in the second stage, thereby keeping the image size small.

Reply:
Got it 👍